src/main/java/hudson/cli/CLICommandInvoker.java

```diff
@@ -121,7 +128,11 @@ public Result invoke() {
                 args, locale, stdin, new PrintStream(out), new PrintStream(err)
         );
 
-        return new Result(returnCode, out, err);
+        Result result = new Result(returnCode, out, err);
+
+        restoreAuth();
```

Comment:
A little safer to do this in a `finally` block. `main` does catch `Exception`, though, so probably good enough (any `Error` is most likely going to fail the whole test case anyway).